backend/de.metas.acct.base/src/main/sql/postgresql/system/43-de.metas.acct/5757940_sys_TaxReport_SupportAccounts_Based_on_AccountConceptualName.sql

```diff
@@ -0,0 +1,126 @@
+                      (CASE
+                           WHEN gll.dr_autotaxaccount = 'Y'
+                               THEN gll.dr_taxbaseamt
+                           WHEN cr_autotaxaccount = 'Y'
+                               THEN gll.cr_taxbaseamt::numeric
+                       END)
+                  WHEN fa.ad_table_id = get_Table_Id('SAP_GLJournal')   THEN
+                      (SELECT amtacct
+                       FROM SAP_GLJournalLine gll
+                       WHERE gll.sap_gljournalline_id = sap_gll.parent_id
+                         AND sap_gll.c_tax_id = gll.c_tax_id)::numeric
+              END)                         AS TaxBaseAmt,
+             c.iso_code                    AS source_currency,
+             cr.iso_code                   AS currency,
+             fa.DocBaseType,
+             fa.C_Tax_ID,
+             fa.account_id,
+             fa.postingtype,
+             fa.accountconceptualname,
+             fa.c_currency_id              AS source_currency_id,
+             cr.c_currency_id,
+             fa.ad_table_id,
+             fa.record_id,
+             fa.ad_org_id,
+             fa.ad_client_id
+      FROM fact_acct fa
+               INNER JOIN c_elementvalue ev ON ev.c_elementvalue_id = fa.account_id
+               INNER JOIN c_tax tax ON fa.c_tax_id = tax.c_tax_id
+               INNER JOIN C_Currency c ON fa.c_Currency_ID = c.C_Currency_ID
+               LEFT OUTER JOIN c_bpartner bp ON fa.c_bpartner_id = bp.c_bpartner_id
+               LEFT OUTER JOIN C_InvoiceTax inv_tax ON (fa.record_id = inv_tax.c_invoice_id AND fa.ad_table_id = get_Table_Id('C_Invoice') AND inv_tax.c_tax_id = fa.c_tax_id)
+               LEFT OUTER JOIN GL_JournalLine gll ON (fa.record_id = gll.gl_journal_id AND fa.ad_table_id = get_Table_Id('GL_Journal') AND gll.gl_journalline_id = fa.line_id AND COALESCE(gll.dr_tax_id, gll.cr_tax_id) = fa.c_tax_id)
+               LEFT OUTER JOIN SAP_GLJournalLine sap_gll ON (fa.record_id = sap_gll.SAP_GLJournal_ID AND fa.ad_table_id = get_Table_Id('SAP_GLJournal') AND
+                                                             sap_gll.SAP_GLJournalLine_ID = fa.line_id AND sap_gll.C_tax_id = fa.c_tax_id)
+               LEFT OUTER JOIN c_allocationline al ON (al.c_allocationline_id = fa.line_id AND al.c_allocationhdr_id = fa.record_id AND fa.ad_table_id = get_Table_Id('C_AllocationHdr'))
+               INNER JOIN AD_ClientInfo ci ON ci.AD_Client_ID = fa.ad_client_id
+               INNER JOIN C_AcctSchema acs ON acs.C_AcctSchema_ID = ci.C_AcctSchema1_ID
+               INNER JOIN C_Currency cr ON acs.C_Currency_ID = cr.C_Currency_ID) AS fa
+         LEFT OUTER JOIN tax_accounts ON (tax_accounts.C_ElementValue_ID = fa.account_id AND fa.ad_table_id IN (get_Table_Id('SAP_GLJournal'), get_Table_Id('GL_Journal')))
+WHERE (fa.ad_table_id IN (get_Table_Id('SAP_GLJournal'), get_Table_Id('GL_Journal')) AND tax_accounts.C_ElementValue_ID IS NOT NULL)
```

Comment:
what a twisted way of writing this condition?! but fine for now, not from this issue